### PR TITLE
Add a constraint to limit DAC

### PR DESCRIPTION
**Pull request by tgilon**
Closes https://github.com/open-energy-transition/form-energy-storage/issues/41.

## Changes proposed in this Pull Request

Suggest a constraint to limit DAC usage.

The current model configuration considers only a subset of sectors. The fuel shares of the transport demand are defined exogenously. This leads to incompressible emissions from land transport oil (366 MtCO2) and associated oil refining (19 Mt CO2). The current CO2 limit is 371 Mt CO2,  which is lower than the land transport emissions (385 Mt CO2). This represents a gap of 14 MtCO2.

The current scenario must be designed so that the CO2 from DAC is not used by Sabatier to produce e-methane. Of the 130 MtCO2 currently captured by DAC, 39 MtCO2 is used by Sabatier and the rest is sequestered (100 Mt CO2).

Completely disabling DAC is not an option because of the transport emissions. This PR proposes to limit the CO2 captured by DAC to 20 MtCO2 (slightly more than the 14 Mt CO2 gap).

A 20 MtCO2 limit for DAC is optimistic but relates to the [European Commission Industrial Carbon Management (COM/2024/62)](https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=CELEX:52024DC0062).

image

A 50 MtCO2 target is given for 2030 and 280 MtCO2 for 2040. Interpolating this for 2035 gives 165 MtCO2. As 25% of it is projected to be capture by DAC in 2040, a 12.5% target is assumed in 2035. This gives a 20.6 MtCO2 target for 2035 ( `(50 + (280 - 50) / 2) * 0.25 / 2)`).

We are aware that is a raw approximation giving only an order of magnitude. Geographical scopes are not aligned neither as the current version of the model doesn't consider all the EU countries.

[European Commission Framework for Carbon removals](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=COM%3A2022%3A672%3AFIN&qid=1669907104132) also gives a lower bound of 5MtCO2 being removed using industrial technologies by 2030.

# ToDo
- [x] Avoid hard coded values, define a configuration
- [x] Improve documentation of the code

## Checklist

- [x] I tested my contribution locally and it works as intended.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in `config/config.default.yaml`.
- [x] Changes in configuration options are documented in `doc/configtables/*.csv`.
- [ ] Sources of newly added data are documented in `doc/data_sources.rst`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

**Comment by daniel-rdt** [comment]
I added the configuration option and documentation for the DAC limit constraint. Maybe we want to amend this to include a pathway? @tgilon

File: config/config.form.yaml [hunk 1]
```diff
@@ -180,6 +180,8 @@ sector:
   reduce_space_heat_exogenously_factor: # TODO: check this for Germany (Regulation or Ariadne?)
     2035: 0.11
   tes: false
+  dac: true
+  dac_limit: false  # False or limit in MtCO2
```
Comment (daniel-rdt) [inline config/config.form.yaml]: atm, this is specified as either false or the limit in MtCO2. An alternative could be to include multiple levels such as `enable` and then optimization years.
```suggestion
  dac_limit:
      enable: False
      2035: 20  # limit in MtCO2
```

**Comment by tgilon** [comment]
> I added the configuration option and documentation for the DAC limit constraint. Maybe we want to amend this to include a pathway? @tgilon

It definitely makes sense to do so for DAC.

File: config/config.form.yaml (hunk 1 above)
Comment (tgilon) [reply config/config.form.yaml]: I like this one !

File: config/config.default.yaml [hunk 2]
```diff
@@ -615,6 +615,7 @@ sector:
   methanation: true
   coal_cc: false
   dac: true
+  dac_limit: false  # false or limit in MtCO2
```
Comment (tgilon) [inline config/config.default.yaml]:
```suggestion
  dac_limit:
      enable: False
      2035: 20  # limit in MtCO2
```

File: doc/configtables/sector.csv [hunk 3]
```diff
@@ -105,6 +105,7 @@ storage_units,--,List of storage technologies to include as StorageUnit componen
 methanation,--,"{true, false}",Add option for transforming hydrogen and CO2 into methane using methanation.
 coal_cc,--,"{true, false}",Add option for coal CHPs with carbon capture
 dac,--,"{true, false}",Add option for Direct Air Capture (DAC)
+dac_limit,--,float or false,Add limit on Direct Air Capture (DAC). Can be either false or limit in MtCO2.
```
Comment (tgilon) [inline doc/configtables/sector.csv]: This need to be updated

File: doc/release_notes.rst [hunk 4]
```diff
@@ -11,6 +11,8 @@ Release Notes
 Upcoming Release
 ================
 
+* Added option to include an exogenous DAC limit specified via the configuration file
```
Comment (tgilon) [inline doc/release_notes.rst]:
```suggestion
* Added option to include an exogenous DAC yearly limit specified via the configuration file (in MtCO2)
```

File: scripts/solve_network.py [hunk 5]
```diff
@@ -966,6 +976,8 @@ def extra_functionality(n, snapshots):
     add_battery_constraints(n)
     add_lossy_bidirectional_link_constraints(n)
     add_pipe_retrofit_constraint(n)
+    if config["sector"]["dac"] and config["sector"]["dac_limit"]:
```
Comment (tgilon) [inline scripts/solve_network.py]: Need to be updated as we use pathways

File: scripts/solve_network.py [hunk 6]
```diff
@@ -931,6 +931,16 @@ def add_co2_atmosphere_constraint(n, snapshots):
             n.model.add_constraints(lhs <= rhs, name=f"GlobalConstraint-{name}")
 
 
+def add_dac_limit(n):
+    """
+    Limit the amount of CO2 captured using DAC. The constraint is formulated in tCO2.
+    """
+    dac = n.links.query("Link.str.contains('DAC')")
+    lhs = (n.model["Link-p"].loc[:, dac.index] * n.snapshot_weightings.generators * dac.efficiency2 * -1).sum()
+    rhs = n.config["sector"]["dac_limit"] * 1e6  # DAC limit specified in config in MtCO2
```
Comment (tgilon) [inline scripts/solve_network.py]: Need to be updated as we use pathways

**Review by tgilon — COMMENTED** [review]
A few changes still required to use pathways

File: doc/configtables/sector.csv (hunk 3 above)
Comment (daniel-rdt) [reply doc/configtables/sector.csv]:
```suggestion
dac_limit,MtCO2,Dictionary with planning horizons as keys.,Limit on Direct Air Capture (DAC) in MtCO2.
-- enable,--,"{true, false}",Add limit on Direct Air Capture (DAC).
```

**Pull request merged**